internal/storage/storageutil/client.go

```diff
@@ -76,39 +77,40 @@ func CreateHttpClient(storageClientConfig *StorageClientConfig) (httpClient *htt
 		}
 	}
 
-	tokenSrc, err := CreateTokenSource(storageClientConfig)
-	if err != nil {
-		err = fmt.Errorf("while fetching tokenSource: %w", err)
-		return
-	}
-
-	// Custom http client for Go Client.
-	httpClient = &http.Client{
-		Transport: &oauth2.Transport{
-			Base:   transport,
-			Source: tokenSrc,
-		},
-		Timeout: storageClientConfig.HttpClientTimeout,
-	}
+	if storageClientConfig.SkipAuth {
+		httpClient = &http.Client{
+			Timeout: storageClientConfig.HttpClientTimeout,
+		}
+	} else {
+		var tokenSrc oauth2.TokenSource
+		tokenSrc, err = CreateTokenSource(storageClientConfig)
+		if err != nil {
+			err = fmt.Errorf("while fetching tokenSource: %w", err)
+			return
+		}
 
-	// Setting UserAgent through RoundTripper middleware
-	httpClient.Transport = &userAgentRoundTripper{
-		wrapped:   httpClient.Transport,
-		UserAgent: storageClientConfig.UserAgent,
+		// Custom http client for Go Client.
+		httpClient = &http.Client{
+			Transport: &oauth2.Transport{
+				Base:   transport,
+				Source: tokenSrc,
+			},
+			Timeout: storageClientConfig.HttpClientTimeout,
+		}
+		// Setting UserAgent through RoundTripper middleware
+		httpClient.Transport = &userAgentRoundTripper{
```

Comment:
Why are we removing user-agent, when skip-auth is true? Ideally, we should only remove tokenSrc part in no-auth scenario.

Reply:
Thanks, I had understanding that tokenSrc is required for that as it added under transport.

Reply:
If we add Transport then it will have source: Token. If we pass token nil then while mounting it fails with `oauth2: Transport's Source is nil`.  So we can't add Transport while creating http client.

Reply:
@tritone is this expected behavour even if we pass WithOutAuthentication option?

Reply:
Checked with @codyoss , In case of defining transport in http client, it will override WithoutAuthentication option.

Reply:
WithUserAgent returns a ClientOption that sets the User-Agent. This option is incompatible with the WithHTTPClient option.

And we are setting client with `WithHTTPClient` option.

We can do this via RoundTripper, but it requires oAuth2.Transport initialized in http client with tokenSrc.